### ISSUE TITLE
Experimental WSL does not support .env file

### ISSUE THREAD

**Issue by enjibby** [microsoft/vscode-remote-release#2754]
- VSCode Version: 1.44
- Local OS Version: 2004 build 19041.173
- Remote OS Version: Ubuntu 19.10 (WSL2)
- Remote Extension/Connection Type: Docker/WSL

Steps to Reproduce:

1. Enable "Experimental WSL" in VS-Code settings
2. Open a project that includes a devcontainer.json and an .env file in the root using the \\\\wsl$\ folder source
3. Use the notification to "Re-open in a container"
4. Note that the environment variables that would be available to docker-compose if starting from a Windows folder are not so in \\\\wsl$\\

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

**Comment by enjibby** [comment]
I know this issue is a bit bleeding-edge and on experimental behaviour. I felt it worth noting as it would allow me to speed up my processes significantly, especially given https://github.com/microsoft/WSL/issues/4197#issuecomment-604592340

**Comment by chrmarti** [comment]
I can forward an environment variable from `.env` to the container using `environment` in the docker-compose.yml:
```yml
version: '3'
services:
  web:
    ...
    environment:
      - FOO
```

Note that the `.env` file needs to be in the project's root folder.

Could you check if that works? What are your config files if it doesn't?

**Comment by chrmarti** [comment]
Actually, you're right, we don't read COMPOSE_PROJECT_NAME from .env correctly. Fixed as part of #2779.
